src/Farmer/Builders/Builders.WebApp.fs

```diff
@@ -364,6 +377,16 @@ type WebAppConfig =
                   Tags = this.Tags}
             | _ ->
                 ()
+
+            yield! this.SiteExtensions
+                |> List.map (fun x ->
+                    {
+                        SiteName = this.Name
+                        Name     = ResourceName x.Name
+                        Location = match x.Location with
+                                    | None -> location // Default to main location
+                                    | Some loc -> loc
+                    } :> IArmResource ) // Can't understand why I have to *upcast* . An anomaly of trying to mix OO and FP? Compiler insists that I be clear on what type I'm working with?
```

Comment:
Yes, this is an F# thing. It doesn't automatically upcast types *within generics e.g. lists*.

Reply:
It makes a kind of sense. I can live with it :-)

Reply:
I *believe* that if you switch to using `[ for x in this.SiteExtensions do .... ]` syntax, you won't need the upcast.